File: 2021/go/day2.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func Day2(input []string) int {

	x := 0
	y := 0
	aim := 0
	for _, v := range input {
		res := strings.Split(v, " ")
		dir := res[0]
		num, _ := strconv.Atoi(res[1])
		if dir == "forward" {
			x += num
			y += num * aim
		} else if dir == "up" {
			// y += num;
			aim -= num
		} else if dir == "down" {
			// y -= num;
			aim += num
		}
	}

	return x * y
}

func main() {
	text, err := os.ReadFile("./../input/day2.txt")
	if err != nil {
		fmt.Println(err)
		return
	}
	input := strings.Split(string(text), "\n")

	fmt.Println(Day2(input))
}

File: 2021/js/day2.js
let fs = require("fs");
let path = require("path");

let input = fs
  .readFileSync(path.resolve(__dirname, "./../input/day2.txt"), "utf8")
  .split("\n");

const getPosition = (arr) => {
  arr = arr.map((el) => el.split(" "));
  let [x, y] = [0, 0];
  let aim = 0;

  for (let i = 0; i < arr.length; i++) {
    const [dir, num] = arr[i];
    if (dir === "forward") {
      x += parseInt(num);
      y += parseInt(num) * aim;
    } else if (dir === "up") {
      // y += parseInt(num);
      aim -= parseInt(num);
    } else if (dir === "down") {
      // y -= parseInt(num);
      aim += parseInt(num);
    }
  }

  return x * y;
};

console.log(getPosition(input));
